graphql_api/types/coverage_analytics/coverage_analytics.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Iterable, Optional, Union
+
+from ariadne import ObjectType, UnionType
+from graphql.type.definition import GraphQLResolveInfo
+
+import timeseries.helpers as timeseries_helpers
+from codecov.db import sync_to_async
+from core.models import Repository
+from graphql_api.types.errors.errors import NotFoundError
+from timeseries.helpers import fill_sparse_measurements
+from timeseries.models import Interval, MeasurementSummary
+
+log = logging.getLogger(__name__)
+
+# Bindings for GraphQL types
+coverage_analytics_bindable: ObjectType = ObjectType("CoverageAnalytics")
+coverage_analytics_result_bindable: UnionType = UnionType("CoverageAnalyticsResult")
+
+
+# CoverageAnalyticsProps is information passed from parent resolver (repository)
+# to the coverage analytics resolver
+@dataclass
+class CoverageAnalyticsProps:
```

Comment:
I propose this pattern of declaring the type of the "parent" that is the first argument to any ariadne resolver with name of `<currentType>Props`. Takes a page out of React playbook and indicates it is a collection of strongly-typed stuff passed directly from parent to child.

The parent (repository) passes this information as below, and the return type of CoverageAnalyticsProps is clear and discoverable
```
@repository_bindable.field("coverageAnalytics")
def resolve_coverage_analytics(
        repository: Repository,
        info: GraphQLResolveInfo,
) -> CoverageAnalyticsProps:
    return CoverageAnalyticsProps(
        repository=repository,
    )
```

Reply:
I was thinking there would be a configuration option to disable default resolvers, but for the life of me I can not find any way to do it on the Ariadne docs.

What you proposed sounds like a good alternative to me, though I'm imagining it would take a bit for devs to build the muscle memory to add follow this guideline.

Reply:
what would you propose for all our existing patterns? I can see a world where we do this for a few models but never go around to cleaning up the rest.

For that reason, I'd say we should probably look to create a proposal outside of this review and gain consensus / context across the team so anyone who happens to miss this review is still informed with the new pattern going forward, and we have a gameplan and bandwidth allocated to actually do this migration across the board as well.

Off the top of your head, would you consider this to take a single engineer a week, a sprint, etc?

Reply:
Thanks so much for your input here and in our Applications Weekly meeting!!
As we discussed, I'll merge this as a trial run of this pattern and we can adjust as we mature team opinions on this. 

re: How long to do migrations across the board - uhhh I'd guess it can be on the order of a sprint depending on how familiar the given engineer is already with `api`, but I'd be more concerned that it's risky, so I'd want to be more sure we want this before touching all those.

Great suggestion there - I spun off [this ticket](https://github.com/codecov/engineering-team/issues/2598) to track some consensus investigation work